doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,456 @@
+  }
+
+  /**
+  * Populates the plan with a path to the given joint space goal
+  * Returns the success of planning
+  **/
+  bool planJointSpaceGoal(const std::vector<double>& joint_group_positions,
+                          moveit::planning_interface::MoveGroupInterface::Plan& plan)
+  {
+    // Under some circumstances the robot may be unable to move as the plan's start state
+    // isn't where the actual robot is. These can be used to update this
+    /* move_group_.setStartState(*move_group_.getCurrentState()); */
+    /* start_state.setFromIK(joint_model_group_); */
+
+    // BEGIN_SUB_TUTORIAL plan_joint_space
+    // To plan to the goal we must set move_group's target to our goal
+    move_group_.setJointValueTarget(joint_group_positions);
+    // move_group can now plan to this goal
+    bool success = (move_group_.plan(plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    // END_SUB_TUTORIAL'
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan (joint space goal) %s", success ? "" : "FAILED");
+    return success;
+  }
+
+  /**
+  * Populates the plan with a path to the pose goal
+  * Returns the success of planning
+  **/
+  bool planPoseGoal(const geometry_msgs::Pose& pose, moveit::planning_interface::MoveGroupInterface::Plan& plan,
+                    bool resetStartState = false)
+  {
+    // Someitmes it can be useful to plan a path that doesn't start where the current robot is. If that is done it is
+    // important to reset the planning start state back to where the robot is when done. This is also used when an
+    // object is attached.
+    if (resetStartState)
+      move_group_.setStartState(*move_group_.getCurrentState());
+
+    // BEGIN_SUB_TUTORIAL plan_pose_goal
+    // Setting the target for a pose goal is similar to a joint space goal
+    move_group_.setPoseTarget(pose);
+    // Planning to the goal, and then moving is identical to the joint space goal.
+    //
+    // END_SUB_TUTORIAL
+    bool success = (move_group_.plan(plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan (pose space goal) %s", success ? "" : "FAILED");
+    return success;
+  }
+
+  /**
+  * Adds the given planning constraints
+  **/
+  void addConstraint(const moveit_msgs::Constraints& test_constraints)
+  {
+    // BEGIN_SUB_TUTORIAL add_constraint
+    // Add it to move_group
+    move_group_.setPathConstraints(test_constraints);
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Removes planning constraints
+  **/
+  void removeConstraints()
+  {
+    // BEGIN_SUB_TUTORIAL remove_constraint
+    // After moving it is often useful to revert back to the default settings.
+    move_group_.clearPathConstraints();
+    // The demo should show two paths to compare constrained plan with an unconstrained plan.
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+   * This sets the planning time to a number of seconds.
+   **/
+  void setPlanningTime(double time = 5.0)
```

Comment:
why are we wrapping single function calls with a function? this function provides no value other than adding complexity to what should be a simple tutorial